votor/src/certificate_pool.rs

```diff
@@ -697,10 +698,15 @@ impl CertificatePool {
         self.completed_certificates
             .iter()
             .filter_map(|(_, cert)| {
-                if Some(cert.certificate.slot()) >= self.highest_finalized_with_notarize {
-                    Some(cert.clone())
-                } else {
-                    None
+                match Some(cert.certificate.slot()).cmp(&self.highest_finalized_with_notarize) {
+                    Ordering::Greater => Some(cert.clone()),
+                    Ordering::Equal => match cert.certificate.certificate_type() {
```

Comment:
nit: Not a huge deal but if we have a Notarze, Finalize, & Fast finalize for the slot we should only send the Fast Finalize, this code will send all 3.

Reply:
Maybe it would be easier if `highest_finalized_with_notarize` also stored the certificate and then we only filtered by > here?